Mathlib/CategoryTheory/Comma/OverClass.lean

```diff
@@ -108,7 +108,32 @@ instance [OverClass X S]
     [IsOverTower X S S'] [IsOverTower Y S S'] [HomIsOver f S] : HomIsOver f S' :=
   homIsOver_of_isOverTower f S S'
 
+variable (X) in
 /-- Bundle `X` with an `OverClass X S` instance into `Over S`. -/
 def OverClass.asOver [OverClass X S] : Over S := Over.mk (X ↘ S)
 
+@[simp]
+lemma OverClass.asOver_hom [OverClass X S] :
+    (OverClass.asOver X S).hom = X ↘ S :=
+  rfl
+
+/-- Bundle a morphism `f : X ⟶ Y` with `HomIsOver f S` into a morphism in `Over S`. -/
+def OverClass.asOverHom [OverClass X S] [OverClass Y S] (f : X ⟶ Y) [HomIsOver f S] :
+    OverClass.asOver X S ⟶ OverClass.asOver Y S :=
+  Over.homMk f (comp_over f S)
+
+@[simp]
+lemma OverClass.asOverHom_left [OverClass X S] [OverClass Y S] (f : X ⟶ Y) [HomIsOver f S] :
+    (OverClass.asOverHom S f).left = f :=
+  rfl
+
+instance OverClass.fromOver {S : C} (X : Over S) : OverClass X.left S where
+  hom := X.hom
+
+@[simp]
+lemma OverClass.fromOver_hom {S : C} (X : Over S) : X.left ↘ S = X.hom := rfl
```

Comment:
Here, it would be better to do `@[simps] instance OverClass.fromOver`. This generates the same lemma as `fromOver_hom`, but with the name `fromOver_over` which should be the more correct name because of the line `initialize_simps_projections OverClass (hom → over)` above in the same file.